core/ops_builtin_v8.rs

```diff
@@ -874,6 +879,73 @@ pub fn op_apply_source_map_filename(
     .ok_or_else(|| type_error("No stashed file name"))
 }
 
+#[op2(core)]
+#[string]
+pub fn op_current_user_call_site(
+  scope: &mut v8::HandleScope,
+  js_runtime_state: &mut JsRuntimeState,
+  #[buffer] ret_buf: &mut [u8],
+) -> String {
+  let stack_trace = v8::StackTrace::current_stack_trace(scope, 10).unwrap();
```

Comment:
This might be missing async stack frames, we do the whole `prepareStackTrace()` thing in `JsError::from_exception()` because of that.

Reply:
Oh! I couldn't remember the reason why we did that, but yeah, it's the async stack frames

Reply:
That's actually fine for my uses. It's ok to say that it will capture the first _sync_ user call site above the call. The use case I have is `Deno.test`'s location finder, which in both `Deno.test` and `t.step()` will have the user's call site synchronously in a stack frame above the current one. This works for `t.step()`, because we capture the stack trace before suspending at the first `await` in the function.

Reply:
Could you verify it that it still works if you have a top-level await before a `Deno.test` declaration?